efel/pyfeatures/pyfeatures.py

```diff
@@ -51,6 +54,27 @@ def time():
     return _get_cpp_feature("time")
 
 
+def impedance():
+    volta = _get_cpp_feature("voltage")
```

Comment:
When t impedance test this is run, this line is reading some other voltage, not the one specified by the test.
What might be the reason of it? @wvangeit  @DrTaDa

Reply:
Did you reset efel, using 'efel.reset()' before running the test?

Reply:
Yes I tried that. It still reads the old voltage.

Reply:
Ok, and what do you mean with the 'old' voltage ? Is it the voltage from another test?

Reply:
I think so. The voltage input data Maria gave me has 52000 elements. The voltage retrieved while running the test only has 53 elements.

Reply:
Just be careful though, there is an interpolation step happening. Is the time trace in the correct units (ms) ? Because if not it might mess up the interpolation

Reply:
Do you mean the unit of time trace in the test input data?

Reply:
yeah, it could be s. It 'might' explain the drop from 52000 -> 53 elements, but I could be wrong

Reply:
I see. Here is what I did to create the test data:
1. From the input `ibw` file read into `c_header` and `c_signal`, `c_header.dx` is 0.0001 and `c_signal.shape[0]` is 52000
2. I wrote this code to generate the time signal
 `t_signal = np.arange(0, c_signal.shape[0] * c_header.dx , c_header.dx)`
3. Then I used `np.savetxt()` with the default value for formatting ` fmt='%.18e'`

Reply:
so dt is 0.0001 ? That seems to suggest that the data is in seconds. If you transform this in ms, things might be fine

Reply:
Yes it was the interpolation. Now it works! 
But I still didn't understand why the interpolation is shrinking the input dimensions?

Reply:
The interpolation step is sampling the data every 0.1 ms and it assumes the time is in milliseconds. If the units are 'wrong', it will sample every 0.1 s, so in this case you loose thousands of points.